addons/web/static/tests/views/fields/one2many_field_tests.js

```diff
@@ -12516,4 +12516,35 @@ QUnit.module("Fields", (hooks) => {
         assert.containsNone(target, ".o_list_record_remove");
         assert.hasClass(target.querySelector(".o_data_row:first-child"), "o_selected_row");
     });
+
+    QUnit.test('Add a line, click on "Save & New" with an invalid form', async function (assert) {
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <field name="p">
+                        <tree>
+                            <field name="display_name"/>
+                        </tree>
+                        <form>
+                            <field name="display_name" required="1"/>
+                        </form>
+                    </field>
+                </form>`,
+        });
+
+        assert.containsNone(target, ".o_data_row");
+        // Add a new record
+        await addRow(target);
+
+        // Click on "Save & New" with an invalid form 
+        await click(target, ".o_form_button_save_new");
+
+        // Check that no buttons are disabled
+        assert.hasAttrValue(target.querySelector(".o_form_button_save"), "disabled", undefined);
```

Comment:
here you should add ".o_dialog" in the selectors because what you want to check is the buttons of the dialog (so actually it's twice, not 3 times):
"o_form_button_save" isn't in the dialog (?) -> can be removed
"o_form_button_save_new" is only in the dialog -> was fine, but cost nothing to have a more precise selector
"o_form_button_cancel" is both behind and in the dialog -> here you do not check what you think you do

Reply:
The check on `o_form_button_save` was because in the modal we have 3 buttons, and this one corresponds to the "Save & Close" button, which was also disabled.